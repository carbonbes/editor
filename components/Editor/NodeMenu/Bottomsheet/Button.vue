<template>
  <button class="h-12 flex items-center justify-between" @click="onClick">
    <div class="flex items-center gap-4 [&>.iconify]:!size-7 ">
      <slot />
    </div>

    <Icon v-if="nextPageTrigger" name="tabler:chevron-right" class="!size-5" />
  </button>
</template>

<script setup lang="ts">
import { injectDrawerContext } from '~/components/Shared/Drawer/DrawerRoot.vue'

const { nextPageTrigger, preventClose } = defineProps<{ nextPageTrigger?: boolean, preventClose?: boolean }>()

const { open } = injectDrawerContext()

function onClick() {
  if (preventClose) return

  open.value = false
}
</script>
