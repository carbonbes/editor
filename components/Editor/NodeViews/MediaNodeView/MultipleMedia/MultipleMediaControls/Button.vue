<template>
  <BaseButton
    as="button"
    class="p-1.5 bg-white shadow rounded-xl cursor-pointer [&>.iconify]:!size-6"
  >
    <slot />
  </BaseButton>
</template>

<script setup lang="ts">
import BaseButton from '~/components/Shared/Base/BaseButton.vue'
</script>
