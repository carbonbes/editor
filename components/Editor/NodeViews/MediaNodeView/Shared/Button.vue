<template>
  <BaseButton class="p-1.5 bg-white rounded-xl shadow">
    <slot />
  </BaseButton>
</template>

<script setup lang="ts">
import { BaseButton } from '~/components/Shared/Base'
</script>
