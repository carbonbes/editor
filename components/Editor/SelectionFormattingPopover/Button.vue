<template>
  <button
    class="p-1.5 flex items-center justify-center rounded-xl cursor-pointer hover:bg-gray-100 [&>.iconify]:!size-5"
    :class="{ 'text-blue-500 !bg-blue-50': active }"
  >
    <slot />
  </button>
</template>

<script setup lang="ts">
const { active } = defineProps<{ active: boolean }>()
</script>
