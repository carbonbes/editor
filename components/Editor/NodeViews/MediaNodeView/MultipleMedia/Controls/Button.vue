<template>
  <Button class="[&>.iconify]:!size-6">
    <slot />
  </Button>
</template>

<script setup lang="ts">
import Button from '~/components/Editor/NodeViews/MediaNodeView/Shared/Button.vue'
</script>
