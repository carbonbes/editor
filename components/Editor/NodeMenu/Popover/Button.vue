<template>
  <BaseFloating
    as="button"
    class="p-1.5 flex items-center justify-center rounded-full cursor-pointer hover:bg-gray-100"
    :class="{ '!bg-gray-200': active }"
  >
    <slot />
  </BaseFloating>
</template>

<script setup lang="ts">
import { BaseFloating } from '~/components/Shared/Base'

const { active } = defineProps<{ active: boolean }>()
</script>
