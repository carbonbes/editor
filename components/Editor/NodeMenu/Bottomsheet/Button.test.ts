import { describe, expect, test } from 'vitest'
import Button from './Button.vue'
import mountDrawer from '~/tests/utils/mountDrawer'

const drawerTemplate = `
  <DrawerBody>
    <DrawerContent>
      <slot />
    </DrawerContent>
  </DrawerBody>
`

function createButtonComponent(template: string) {
  return defineComponent({
    components: { Button },
    template,
  })
}

describe('Button', () => {
  test('should be defined', () => {
    expect(Button).toBeDefined()
  })

  describe('is displayed correctly', () => {
    const cases = [
      ['without a slot', '<Button></Button>'],
      ['with a slot', '<Button><p>Test</p></Button>'],
      ['with an chevron-right icon', '<Button :next-page-trigger="true"><p>Test</p></Button>'],
    ]

    test.each(cases)('%s', async (_, buttonTemplate) => {
      const ButtonComponent = createButtonComponent(buttonTemplate)
      const drawer = await mountDrawer({ template: drawerTemplate, slot: ButtonComponent })
      const button = drawer.getComponent(Button)

      expect(button.html()).toMatchSnapshot()

      drawer.unmount()
    })
  })

  test('pressing should close the Drawer', async () => {
    const ButtonComponent = createButtonComponent('<Button></Button>')
    const drawer = await mountDrawer({ template: drawerTemplate, slot: ButtonComponent })
    const button = drawer.getComponent(Button)

    await button.trigger('click')

    expect(drawer.vm.open).toBeFalsy()

    drawer.unmount()
  })
})
