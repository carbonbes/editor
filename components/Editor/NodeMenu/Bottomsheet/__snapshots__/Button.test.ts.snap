// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > is displayed correctly > with a slot 1`] = `
"<button class="h-12 flex items-center justify-between">
  <div class="flex items-center gap-4 [&amp;>.iconify]:!size-7">
    <p>Test</p>
  </div>
  <!--v-if-->
</button>"
`;

exports[`Button > is displayed correctly > with an chevron-right icon 1`] = `
"<button class="h-12 flex items-center justify-between">
  <div class="flex items-center gap-4 [&amp;>.iconify]:!size-7">
    <p>Test</p>
  </div><span class="iconify i-tabler:chevron-right !size-5" aria-hidden="true"></span>
</button>"
`;

exports[`Button > is displayed correctly > without a slot 1`] = `
"<button class="h-12 flex items-center justify-between">
  <div class="flex items-center gap-4 [&amp;>.iconify]:!size-7"></div>
  <!--v-if-->
</button>"
`;
